src/StaticWebAssetsSdk/Tasks/ResolveCompressedAssets.cs

```diff
@@ -0,0 +1,266 @@
+            var originalItemSpec = staticWebAsset.GetMetadata("OriginalItemSpec");
+            compressedAssets.Add(CreateCompressedAsset(asset, originalItemSpec));
+        }
+
+        AssetsToCompress = compressedAssets.ToArray();
+
+        return !Log.HasLoggedErrors;
+    }
+
+    private TaskItem CreateCompressedAsset(AssetToCompress asset, string originalItemSpec)
+    {
+        var originalAsset = asset.StaticWebAsset;
+        var relativePath = originalAsset.GetMetadata("RelativePath");
+        var targetDirectory = asset.CompressionConfiguration.TargetDirectory;
+        var format = asset.CompressionConfiguration.Format;
+
+        // TODO: Clean up this method.
+        string fileExtension;
+        string assetTraitValue;
+
+        if (string.Equals("gzip", format, StringComparison.OrdinalIgnoreCase))
+        {
+            fileExtension = ".gz";
+            assetTraitValue = "gzip";
+        }
+        else if (string.Equals("brotli", format, StringComparison.OrdinalIgnoreCase))
+        {
+            fileExtension = ".br";
+            assetTraitValue = "br";
+        }
+        else
+        {
+            fileExtension = ".gz";
+            assetTraitValue = "gzip";
+            Log.LogError($"Unknown compression format '{format}'. Defaulting to 'gzip'.");
+        }
+
+        var outputRelativePath = Path.Combine(
+            targetDirectory,
+            CalculateTargetPath(originalItemSpec, fileExtension));
+
+        var result = new TaskItem(outputRelativePath, originalAsset.CloneCustomMetadata());
+
+        result.SetMetadata("RelativePath", relativePath + fileExtension);
+        result.SetMetadata("RelatedAsset", originalAsset.ItemSpec);
+        result.SetMetadata("OriginalItemSpec", originalAsset.ItemSpec);
+        result.SetMetadata("AssetRole", "Alternative");
+        result.SetMetadata("AssetTraitName", "Content-Encoding");
+        result.SetMetadata("AssetTraitValue", assetTraitValue);
+        result.SetMetadata("TargetDirectory", targetDirectory);
+        result.SetMetadata("Format", format);
+
+        return result;
+    }
+
+    private static string CalculateTargetPath(string relativePath, string extension)
+    {
+        // RelativePath can be long and if used as-is to write the output, might result in long path issues on Windows.
+        // Instead we'll calculate a fixed length path by hashing the input file name. This uses SHA1 similar to the Hash task in MSBuild
+        // since it has no crytographic significance.
+        using var hash = SHA1.Create();
+        var bytes = Encoding.UTF8.GetBytes(relativePath);
+        var hashString = Convert.ToBase64String(hash.ComputeHash(bytes));
+
+        var builder = new StringBuilder();
+
+        for (var i = 0; i < 8; i++)
+        {
+            var c = hashString[i];
+            builder.Append(s_invalidPathChars.Contains(c) ? '+' : c);
+        }
+
+        builder.Append(extension);
+        return builder.ToString();
+    }
```

Comment:
Use `FileHasher` instead to produce a name https://github.com/dotnet/sdk/blob/a821467d6ef6f4d3f9c3de8c468041d8c8d5b3b0/src/WasmSdk/Tasks/FileHasher.cs#L12